src/read_data_falcON_hdf5_utils.c

```
@@ -0,0 +1,551 @@
+        /* try to open the dataset; skip silently if absent */
+        H5E_BEGIN_TRY {
+            dset = H5Dopen2(Parts[type].group_id, field, H5P_DEFAULT);
+        } H5E_END_TRY;
+        if (dset < 0) continue;
+
+        space = H5Dget_space(dset);
+        rank  = H5Sget_simple_extent_ndims(space);
+        H5Sget_simple_extent_dims(space, count, NULL);
+        ndat = (int)Parts[type].number;
+
+        buf = (double *)realloc(buf, (size_t)ndat * sizeof(double));
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)